Add Micrometer AutoConfiguration (#2391)
spring-cloud/spring-cloud-gcp: docs/src/main/asciidoc/metrics.adoc

== Stackdriver Monitoring

Google Cloud Platform provides a service called https://cloud.google.com/monitoring/[Stackdriver Monitoring], and https://micrometer.io/docs/registry/stackdriver[Micrometer] can be used with it to easily instrument Spring Boot applications for observability.

Spring Boot already provides auto-configuration for Stackdriver. This module enables to auto-detect the `project-id` and `credentials`. Also, it can be customized.

Maven coordinates, using <<getting-started.adoc#_bill_of_materials, Spring Cloud GCP BOM>>:

[source,xml]
----
<dependency>
    <groupId>org.springframework.cloud</groupId>
    <artifactId>spring-cloud-gcp-starter-metrics</artifactId>
</dependency>

> ----

Reviewer: Users would also need a dependency on actuator starter; otherwise `Clock` bean will not be autoconfigured, preventing our autoconfig from triggering.

```
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-actuator</artifactId>
    </dependency>
```

Or should this be in the starter, since `spring-cloud-gcp-starter-metrics` is not useful without actuator?

Author: That's right, `actuator` is required. Even though the new starter will not make sense without the `actuator` dependency, I would prefer to let this clear in the docs so the developers are aware about what dependencies are declaring for this case. WDYT?

Reviewer: Yes, I agree. Just add a mention of it to the refdoc.
Otherwise, looks great -- I tested the starter in a tiny sample app.

You must enable Stackdriver Monitoring API from the Google Cloud Console in order to capture metrics.
Navigate to the https://console.cloud.google.com/apis/api/monitoring.googleapis.com/overview[Stackdriver Monitoring API] for your project and make sure it’s enabled.

> This starter depends on `org.springframework.boot:spring-boot-starter-actuator` dependency. Make sure the dependency is being declared.

Reviewer: ```suggestion
This starter requires `org.springframework.boot:spring-boot-starter-actuator` dependency to activate. Make sure the dependency is being declared.
```
